examples/vision/xray_classification_with_tpus.py

```diff
@@ -0,0 +1,479 @@
+`binary_crossentropy` loss. When we fit the model, identify the class weights. Because we
+are using a TPU, training will be relatively quick.
+
+For our metrics, we want to include precision and recall as they will provide use with a
+more informed picture of how good our model is. Accuracy tells us what fractions are the
+labels are correct. Since our data is not balanced, accuracy might give a skewed sense of
+a good model (i.e. a model that always predicts PNEUMONIA will be 74% accurate but is not
+a good model).
+
+Precision is the number of true positives (TP) over the sum of TP and false positives
+(FP). It shows what fraction of labeled positives are actually correct.
+
+Recall is the number of TP over the sum of TP and false negatves (FN). It shows what
+fraction of actual positives are correct.
+"""
+
+with strategy.scope():
+    model = build_model()
+
+    METRICS = [
+        "accuracy",
+        tf.keras.metrics.Precision(name="precision"),
+        tf.keras.metrics.Recall(name="recall"),
+    ]
+
+    model.compile(optimizer="adam", loss="binary_crossentropy", metrics=METRICS)
+
+"""
+From exploring the data and the model, I noticed that the training for the model has a
+slow start. However, after 25 epochs, the model slowly starts to converge.
+"""
+
+"""
+## Finetune the model
+
+Finetuning is an art when it comes to Machine Learning, and there are many ways to adjust
+the model in efforts to improve it. Finetuning is beyond the scope of this notebook, but
+check out this
+[article](https://www.pyimagesearch.com/2019/06/03/fine-tuning-with-keras-and-deep-learnin
+g/) for more information.
+
+For our purposes, we'll use Keras callbacks to further finetune our model. The checkpoint
+callback saves the best weights of the model, so next time we want to use the model, we
+do not have to spend time training it. The early stopping callback stops the training
+process when the model starts becoming stagnant, or even worse, when the model starts
+overfitting. Since we set `restore_best_weights` to `True`, the returned model at the end
+of the training process will be the model with the best weights (i.e. low loss and high
+accuracy).
+"""
+
+checkpoint_cb = tf.keras.callbacks.ModelCheckpoint("xray_model.h5", save_best_only=True)
+
+early_stopping_cb = tf.keras.callbacks.EarlyStopping(
+    patience=10, restore_best_weights=True
+)
+
+"""
+We also want to finetune our learning rate. Too high of a learning rate will cause the
+model to diverge. Too small of a learning rate will cause the model to be too slow. We
+implement the exponential learning rate scheduling method below.
+"""
+
+
+def exponential_decay(lr0, s):
+    def exponential_decay_fn(epoch):
+        return lr0 * 0.1 ** (epoch / s)
+
+    return exponential_decay_fn
+
+
+exponential_decay_fn = exponential_decay(0.015, 20)
+
+lr_scheduler = tf.keras.callbacks.LearningRateScheduler(exponential_decay_fn)
```

Comment:
If you're just doing exponential decay, you can use https://keras.io/api/optimizers/learning_rate_schedules/exponential_decay/
instead of a callback (preferred)